htdocs/core/class/html.form.class.php

```diff
@@ -1925,7 +1925,7 @@ public function select_produits($selected = '', $htmlname = 'productid', $filter
 				}
 			}
 			// mode=1 means customers products
-			$urloption='htmlname='.$htmlname.'&outjson=1&price_level='.$price_level.'&type='.$filtertype.'&mode=1&status='.$status.'&finished='.$finished.'&hidepriceinlabel='.$hidepriceinlabel.'&warehousestatus='.$warehouseStatus;
+			$urloption='htmlname='.$htmlname.'&outjson=1&price_level='.$price_level.'&type='.$filtertype.'&mode=1&status='.$status.'&finished='.$finished.'&hidepriceinlabel='.$hidepriceinlabel.'&warehousestatus='.$warehouseStatus.'&limit='.$limit;
```

Comment:
The parameter limit is already used to save nb of records selected into list. To avoid conflict, i suggest to renama here parameter into  "selectmax" for example.

Reply:
Eldy. Good night. Sorry for my late answer.  I don't understand your request to rename the parameter. In product/ajax/product.php the limit parameter is not defined. The idea about add this parameter is better perf when catalog had thousands of products, but i agree about your idea of a hidden param   PRODUIT_LIMIT_LIST_EVEN_WHEN_USE_SEARCH_TO_SELECT_ON

Reply:
The limit parameter is not defined into product/ajax/product.php but it is in nearly every other pages. And we must avoid conflict of naming in variables for readability of code.
Also the limit must not be on by default because it leads to situation where user is completely blocked (he can't select product and don't know why). So it must be active only if a hidden conf PRODUIT_LIMIT_LIST_EVEN_WHEN_USE_SEARCH_TO_SELECT_ON is set.